tests/integration/conftest.py

```diff
@@ -627,6 +661,35 @@ def qtest_requirements(self):
         except (RuntimeError, AssertionError):
             raise
 
+    @property
+    def qtest_parent_test_cycles(self):
+        """A list of qTest swagger_client test cycle models which are the direct ancestor for all associated test runs.
+
+        Returns:
+            [swagger_client.TestCycleResource]
+
+        Raises:
+            AssertionError: Associated test cycles do not exist.
+            RuntimeError: General qTest API failure.
+        """
+
+        test_cycle_ids = []
+
+        for run in self.qtest_test_runs:
+            for link in run.links:
+                if link.rel == 'test-cycle':
+                    test_cycle_ids.append(int(self._HREF_ID_REGEX.search(link.href).group(1)))
```

Comment:
Is it possible for the search not to find any matches? If so, then the `group` call will throw an `AttributeError`.